content/tutorials/react-native-presence-publish-subscribe.textile

```diff
@@ -0,0 +1,539 @@
+
+```[sh]
+//change directory to ably-presence-publish-subscribe-react-native
+cd ably-presence-publish-subscribe-react-native
+//run the application for android
+react-native run-android
+//run the application for ios
+react-native run-ios
+```
+
+If all goes well, you should see a screen that says “Welcome to React Native” come up on your emulator or device.
+
+
+h2. Setting Up Ably
+
+You need to sign up on "Ably":https://www.ably.io/, create a new app, click on the @manage app@ button at the left-hand side of the app, click on the @Api keys@  tab and copy our Api Key.
+
+<a href="/images/tutorials/copy-api-key.png" target="_blank">
+  <img src="/images/tutorials/copy-api-key.png" style="width: 100%" alt="Copy API Key screenshot">
+</a>
+
+Next, you will install the required libraries for this app. 
+
+```[sh]
+npm install ably ably-react-native express body-parser 
+```
+In the above bash command, you installed 4 packages. I will explain what the four packages are used for:
+
+# express: This is a Node.js web framework which we’ll use to create our API.
+# body-parser: This library is used by Express to parse body requests.
+# ably: This library is the official @Ably@ library for Node js
+# ably-react-native: This library is the official @Ably@ library for React Native
+
+
+h2. Creating Our API
+
+First, Create a new file called @server.js@, which serves as your API in our root folder and paste the following:
+
+```[javascript]
+//require express
+var express = require('express')
+//define app as in instance of express
+var app = express()
+//require bosy-parser
+var bodyParser = require('body-parser')
+
+//use bodyparser as a middle ware
+app.use(bodyParser.json())
+//ably realtime
+const ably = require('ably').Realtime;
+// instantiate an instance of ably passing in our API key
+const myably = new ably('XXX_API_KEY')
+//set the channel you will like to publish to 
+const channel = myably.channels.get('cog-tie');
+//set cors middleware
+app.use(function(req, res, next) {
+res.header("Access-Control-Allow-Origin", "*");
+res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+next();
+});
+//handle route postfunction
+app.post('/', function(req, res) {
+var message = {
+    user: req.body.name,
+    msg: req.body.msg,
+    action: req.body.action
+}
+//Publish your message
+channel.publish('message', message, function(err) {
+    if (err) {
+        console.log('publish failed with error ' + err);
+    } else {
+        console.log('publish succeeded ' + message.msg);
+    }
+})
+res.send({ status: 'okay', message: message })
+
+})
+
+//listen on port and serve the app
+app.listen(3000, function() {
+console.log('Example app listening on port 3000!')
+})
+```
+The code block above is our Express API server setup. 
+
+At the beginning of the file, you had required Express, Body-parser and Ably libraries for Node.js respectively. 
+Next, you initialized a new Pusher object, passing in your Api Key.
```

Comment:
Pusher?

Reply:
I'm so sorry about this. It was a huge mixup. I apologize, and I have changed it accordingly